File: java/json-transform/src/test/java/co/nlighten/jsontransform/functions/TransformerFunctionAndTest.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.BaseTest;
import org.junit.jupiter.api.Test;

public class TransformerFunctionAndTest extends BaseTest {
    @Test
    void assertAnd() {
        var twoAndThree = fromJson("""
                                                   {
                                                   "$$and": [
                                                    { "$$is": "$[0]", "eq": 2 },
                                                    { "$$is": "$[1]", "eq": 3 }
                                                   ]
                                                   }""");
        assertTransformation(fromJson("[2,3]"), twoAndThree, true);
        assertTransformation(fromJson("[2,4]"), twoAndThree, false);
    }

    @Test
    void inline() {
        assertTransformation(fromJson("[null,0]"), "$$and:$", false);
        assertTransformation(fromJson("[1,0]"), "$$and:$", false);
        assertTransformation(fromJson("[1,true]"), "$$and:$", true);
    }
}

File: javascript/json-transform/src/__tests__/functions/TransformerFunctionAnd.test.ts
import { describe, test } from "vitest";
import { assertTransformation } from "../BaseTransformationTest";

describe("TransformerFunctionAnd", () => {
  test("assertAnd", async () => {
    const twoAndThree = {
      $$and: [
        { $$is: "$[0]", eq: 2 },
        { $$is: "$[1]", eq: 3 },
      ],
    };
    await assertTransformation([2, 3], twoAndThree, true);
    await assertTransformation([2, 4], twoAndThree, false);
  });

  test("inline", async () => {
    await assertTransformation([null, 0], "$$and:$", false);
    await assertTransformation([1, 0], "$$and:$", false);
    await assertTransformation([1, true], "$$and:$", true);
  });
});
